### ISSUE TITLE
[PROPOSAL] Add VectorX structs mirroring the System.Numerics Vectors

### ISSUE THREAD

**Issue by ghost** [john-h-k/MathSharp#56]
Numerics Vectors only support floats, we could re-create them for Doubles, ints and longs (and possibly other types?).
This would allow users to take advantage of vector instructions without having to do loads and saves themselves.

**Comment by john-h-k** [comment]
This seems a good idea, but I would keep explicit loads and stores because they have a cost

**Comment by ghost** [comment]
Agreed, the JIT probably optimizes this away with the built-in types but we can't count on it doing the same thing in this case (especially since we'd be explicitly calling the SSE/AVX Load/Store functions).

**Comment by john-h-k** [comment]
I can/will/might do these, but they would just be a storage type, with proper ctors, `ToString/GetHashCode/IEquatable<T>`. No maths on them
